### ISSUE TITLE
Document how to modify resin-wifi-connect

### ISSUE THREAD

**Issue by brownjohnf** [resin-io/resin-wifi-connect#139]
For example, a user was attempting to modify the index.html file, but found that doing so had no affect because the original resin-io version of the repo gets pulled down at build time.

**Comment by dhoney** [comment]
This is exactly what I am trying to do.

**Comment by majorz** [comment]
@dhoney

You may change the location of the assets directory with the `--ui-path` command line option or the `UI_PATH` environment variable.

Let's say you keep your modified index.html in `/some-location/resin-wifi-connect/public`, then you may export `UI_PATH=/some-location/resin-wifi-connect/public` and the internal web server of the application will pick up this location and serve the files from there.

Please let me know if that works for you or if you have any other ideas or suggestions.

**Comment by dhoney** [comment]
@majorz 

Yes I believe that will work for the modified assets.

What about modifications to server.rs?
For example: If I add a field to index.html for storing a file in /data,  I need to modify the connect() function in server.rs to handle that.

**Comment by majorz** [comment]
@dhoney 

Yes, you may extend the connect function: grab the new field content there, process it and store the results in /data. Please let me know if you need help with that, since Rust can be a new language for you.

Additionally the Rust code have to be compiled to produce the binaries.

You can compile locally on Linux to test on x86_64 architecture. But to run in on an ARM device, the cross-compilation toolchain should be used. This is easy now. We use CircleCI for it. Once you fork the repo, you will have to enable CircleCI builds for the forked repo. When you push changes, CircleCI will pick them up and trigger builds for the supported architectures. The resulting binaries can be downloaded from the Artifacts section of each build.

If you want the builds to be uploaded on the GItHub Releases page of the forked repo, then you will have to change this line to point to your GitHub account: https://github.com/resin-io/resin-wifi-connect/blob/master/.circleci/config.yml#L75

In the CircleCI environment variables section for the project you will have to add two environment variables, which are used by the deploy.sh script that uploads the archived binaries to GitHub Releases:
  * GITHUB_TOKEN - which is a GitHub personal access token, that you can generate here: https://github.com/settings/tokens
  * CIRCLE_TOKEN - similar to the above, CircleCI API token, which you may generate here: https://circleci.com/account/api

The deploy script is only triggered when you push annotated git tags to master.

I think there is some generality of your use case, and it will be interesting to see if we can extend the application in future releases, so that it covers it. Like having some command line argument which enables writing all entered fields in a JSON file somewhere in /data. Or sending a request to external server with those fields, etc.

**Comment by dhoney** [comment]
@majorz

I got setup with circleci and it seems to be building the artifact now, thanks!

**Comment by dhoney** [comment]
@majorz 
Curious if you can help me interpret this error I'm getting from cargo when I build

```
error[E0599]: no method named `write_all` found for type `std::fs::File` in the current scope
   --> src/server.rs:235:16
    |
235 |     match file.write_all(uuid.as_bytes())
    |                ^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope, perhaps add a `use` for it:
            candidate #1: `use std::io::Write;`
```

I'm using the file io example as inspiration https://rustbyexample.com/std_misc/file/create.html

My modification is as follows:

```
fn connect(req: &mut Request) -> IronResult<Response> {
    let (ssid, password, uuid) = {
        let params = get_request_ref!(req, Params, "Getting request params failed");
        let ssid = get_param!(params, "ssid", String);
        let password = get_param!(params, "password", String);
        let uuid = get_param!(params, "uuid", String);
        (ssid, password, uuid)
    };

    // Handle writing files
    let uuid_path = Path::new("/data/uuid");
    let uuid_display = uuid_path.display();

    // Open a file in write-only mode, returns `io::Result<File>`
    let mut file = match File::create(&uuid_path) {
        Err(why) => panic!("couldn't create {}: {}",
                           uuid_display,
                           why.description()),
        Ok(file) => file,
    };

    // Write the uuid string to `file`, returns `io::Result<()>`
    match file.write_all(uuid.as_bytes())
    {
        Err(why) => {
            panic!("couldn't write to {}: {}", uuid_display,
                                               why.description())
        },
        Ok(_) => println!("successfully wrote to {}", uuid_display),
    }
```

Been looking through the docs a while, any suggestions appreciated.

**Comment by majorz** [comment]
@dhoney, hey Dan, the compiler suggested adding `use std::io::Write;` so that the implementation of `write_all` is available, but when added like the compiler suggested we run into a name collision, because of `use persistent::Write;`. Using `as` will do the job ;)

```
use std::fs::File;
use std::path::Path;
use std::io::Write as IoWrite;
```

**Comment by dhoney** [comment]
@majorz  hah silly me, thanks Zahari!

**Comment by majorz** [comment]
It is the compiler that needs to get smarter with the suggestions :)))

**Comment by dhoney** [comment]
@majorz last update I promise :D

I got the functionality working so thanks for the assist on that.  
I can now add files to ```/data``` as part of my apps provisioning process.

If you think this is something useful I can post a PR or gist-snippet

**Comment by majorz** [comment]
Yes, please post the snippets once you are ready. Those will be definitely useful once we start to explore how to make the application extendable.

**Comment by drjasonharrison-vp-eio** [comment]
Can the source be built as part of a dockerfile.template? 
@dhoney did you post your gist?
